suite-native/app/e2e/tests/deeplinkPopup.test.ts

```diff
@@ -0,0 +1,165 @@
+                    res.end('Callback URL received successfully!\n');
+                }
+            });
+
+            server.listen(SERVER_PORT, 'localhost', () => {
+                // eslint-disable-next-line no-console
+                console.info(`Server running at ${SERVER_URL}`);
+                resolve(null);
+            });
+        });
+
+        await prepareTrezorEmulator();
+        await openApp({ newInstance: true });
+        await onOnboarding.finishOnboarding();
+
+        await onCoinEnablingInit.waitForScreen();
+        await onCoinEnablingInit.enableNetwork('regtest');
+        await onCoinEnablingInit.clickOnConfirmButton();
+
+        await waitFor(element(by.id('skip-view-only-mode')))
+            .toBeVisible()
+            .withTimeout(10000); // communication between connected Trezor and app takes some time.
+
+        await element(by.id('skip-view-only-mode')).tap();
+
+        // This `TrezorConnect` instance here is pretending to be the integrator or @trezor/connect-mobile
+        await TrezorConnect.init({
+            manifest: {
+                email: 'developer@xyz.com',
+                appUrl: 'http://your.application.com',
+            },
+            deeplinkOpen: url => {
+                openUriScheme(url, 'android');
+            },
+            deeplinkCallbackUrl: `${SERVER_URL}/connect/`,
+            connectSrc: 'https://dev.suite.sldev.cz/connect/develop/',
+        });
+    });
+
+    beforeEach(async () => {
+        await prepareTrezorEmulator();
+        await restartApp();
+
+        await reverseServerPort(SERVER_PORT);
+
+        await appIsFullyLoaded();
+
+        await waitFor(element(by.id('@screen/ConnectingDevice')))
+            .toBeVisible()
+            .withTimeout(10000);
+
+        await onHome.waitForScreen();
+    });
+
+    afterAll(async () => {
+        disconnectTrezorUserEnv();
+        await new Promise(resolve => {
+            if (server) {
+                server.close(() => {
+                    resolve(null);
+                });
+            }
+        });
+        await device.terminateApp();
+    });
+
+    it('Handle deeplink', async () => {
+        const promise = TrezorConnect.getAddress({
+            path: "m/49'/0'/0'/0/0",
+            coin: 'btc',
+        });
+
+        await element(by.id('@popup/deeplink-info'));
+
+        await waitFor(element(by.id('@popup/call-device')))
+            .toBeVisible()
+            .withTimeout(10 * 1000);
+        await element(by.id('@popup/call-device')).tap();
+
+        await TrezorUserEnvLink.pressYes();
+
+        const response = await promise;
+        const expectedResponse = {
+            id: 1,
+            payload: {
+                path: [49, 0, 0, 0, 0],
+                serializedPath: 'm/49/0/0/0/0',
+                address: '3J7UQSLAaFh1nkUomVdm8ArifPEvYfNr1o',
+            },
+            success: true,
+        };
+
+        await new Promise((resolve, reject) => {
```

Comment:
why is the Promise object needed here? What about simplifying it to something like:
```suggestion
        if (JSON.stringify(response) !== JSON.stringify(expectedResponse)) {
                new Error('Result does not match expected.')
            }
```

Reply:
Done simpler https://github.com/trezor/trezor-suite/pull/14667/commits/72c023f4fce281731b15d7681bb6b6e99a6a9912#diff-0cba7b27a14d30c9b8c7c5b3ac0c073b9e94a549ac4c8aeb74012a0676faba91R74

Reply:
This, unfortunately, does not work I tried to put a dummy value to the `expectedResponse` e.g. `{success: false}`, and the test still passed. It is because if the values do not match, an error object is created, but it is not thrown.

To ensure the values are compared correctly, you must use the jest assert function: ' expect (response).toEqual(expectedResponse)`. I am aware that I recommended you to use the Error object in the previous comment, but I was rather focusing on removing the `Promise` object.

Reply:
Sure, let's throw https://github.com/trezor/trezor-suite/pull/14667/commits/407c5d0bb0220465be8e30e83a4515590214958d

`expect` does not work as expected, so I would keep the current approach but throwing the error, I tested with success and fail and it worked.

Reply:
Strange, the expect worked for me locally. I like the `expect` keyword more because it prints out the difference between the two compared values in case of failure. Never mind, this works as well :).